### PR TITLE
Added pubmed and simplified regex in arxiv and piratebay

### PR DESCRIPTION
**Pull request by k4u5h1k**
- Pubmed is a biomedical research repository.
- Added engine and search module for it.
- Removed lookahead, lookbehind portions of regex in arxiv and piratebay.
- Added output string to piratebay prompting usage of magnet link

### CODE REVIEW

**Comment by k4u5h1k** [comment]
fixed critical bug where unset sets to "None" instead of None and causes modules to malfunction

File: core/util/arxiv.py [hunk 1]
```diff
@@ -47,7 +47,7 @@ def run_crawl(self):
                         self.framework.error('ArXiv is missed!')
                         return
                 self._rawxml = req.text
-                self._articles = list(re.findall('(?<=<entry>).*?(?=</entry>)', 
+                self._articles = list(re.findall('<entry>(.*?)</entry>',
```
Comment (saeeddhqan) [inline core/util/arxiv.py]: r"REGEX" do it for other files.

File: core/util/arxiv.py [hunk 2]
```diff
@@ -61,15 +61,15 @@ def articles(self):
 
         @property
         def links(self):
-                self._links = re.findall('http://arxiv.org/abs/.*(?=<)',
+                self._links = re.findall('(http://arxiv.org/abs/.*)<',
```
Comment (saeeddhqan) [inline core/util/arxiv.py]: r""

File: core/util/arxiv.py [hunk 3]
```diff
@@ -61,15 +61,15 @@ def articles(self):
...
                         self._rawxml)
                 return self._links
 
         @property
         def links_with_data(self):
-                findlink = lambda x: re.findall('http://arxiv.org/abs/.*?(?=<)', x)
-                findauthors = lambda x: re.findall('(?<=<name>).*?(?=</name>)', x)
-                findtitle = lambda x: re.findall('(?<=<title>).*?(?=</title>)', x, flags=re.DOTALL)
+                findlink = lambda x: re.findall('(http://arxiv.org/abs/.*?)<', x)
```
Comment (saeeddhqan) [inline core/util/arxiv.py]: r""

File: core/util/pubmed.py [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+                return self._articles
+
+        @property
+        def links_with_data(self):
+                findlink = lambda x: list(map(lambda x: self.pubmed+x, 
+                        re.findall('<a.*?class="docsum-title".*?href="(.*?)".*?>', x, flags=re.DOTALL)))
+                findauthors = lambda x: re.findall('<span class="docsum-authors full-authors">(.*?)</span>', 
+                        x, flags=re.DOTALL)
+                findtitle = lambda x: list(map(lambda x: html.unescape(re.sub('</?b>','',x)).strip(),
+                        re.findall('<a.*?class="docsum-title".*?>(.*?)</a>', x, flags=re.DOTALL)))
```
Comment (saeeddhqan) [inline core/util/pubmed.py]: r""

File: core/util/pubmed.py [hunk 5]
```diff
@@ -0,0 +1,81 @@
...
+                return self._rawhtml
+
+        @property
+        def articles(self):
+                return self._articles
+
+        @property
+        def links_with_data(self):
+                findlink = lambda x: list(map(lambda x: self.pubmed+x, 
+                        re.findall('<a.*?class="docsum-title".*?href="(.*?)".*?>', x, flags=re.DOTALL)))
```
Comment (saeeddhqan) [inline core/util/pubmed.py]: put a prefix: r

**Comment by k4u5h1k** [comment]
done

**Pull request merged**